### PR TITLE
[Board] Adafruit Trinket

### PR DESCRIPTION
**Pull request by Munsio**
Corresponding Issue for discussions: #332

### CODE REVIEW

**Comment by deadprogram** [comment]
At first glance, this looks great @Munsio thank you very much for the contribution.

@aykevl any comments before I merge it?

**Comment by Munsio** [comment]
If you say it looks fine i would like to test it before you merge it. (as stated in the issue i didn't try to run any code on it cause i wasn't sure that it won't brick it)

I get home at aprox. 18:00 GMT and try running the blinky example on it.

**Comment by deadprogram** [comment]
One note: I would use the same instructions to put the code on it with `tinygo build` as the ItsyBitsy. https://tinygo.org/microcontrollers/itsybitsy-m0/#uf2

The UF2 bootloader works quite well and quickly/safely on the various Adafruit boards that support it.

**Comment by deadprogram** [comment]
It is also useful to compare your pin mappings to something like https://github.com/adafruit/ArduinoCore-samd/blob/master/variants/trinket_m0/variant.cpp

**Comment by Munsio** [comment]
I checked it with the file you mentioned and saw that i didn't provide the DotStart and USB-Host Pins - i am not sure if they should be added? The rest seems okay, there are some placeholder pins described which i also not know if they should be included.

**Comment by deadprogram** [comment]
The dotStar requires a bitbanging-style SPI interface, so we do not have that yet.

The USB-host pins you will need, in order to use the serial over USB something like this: https://github.com/tinygo-org/tinygo/blob/master/src/machine/board_itsybitsy-m0.go#L39-L43

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 build
 docs/_build
+llvm*/
```
Comment (aykevl) [inline .gitignore]: While this is a useful change, it doesn't really belong in this PR.

**Comment by Munsio** [comment]
Blinky is working and Serial is working - if you want you can merge it

File: src/machine/board_trinket.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	SDA_PIN = D0 // SDA
+	SCL_PIN = D2 // SCL
+)
+
+// I2C on the Trinket M0.
+var (
+	I2C0 = I2C{Bus: sam.SERCOM2_I2CM,
+		SDA:     SDA_PIN,
+		SCL:     SCL_PIN,
+		PinMode: GPIO_SERCOM}
```
Comment (deadprogram) [inline src/machine/board_trinket.go]: This should be `GPIO_SERCOM_ALT` here, I think.

**Comment by deadprogram** [comment]
I think this PR is about ready to merge, from what I can tell.

Only thing is, it greatly complicates merging my own PR #331 in either case there will be some conflicts to resolve, but I think my changes are fewer, so would better to merge it first, then resolve this one, and merge it.

Any opinions on this?

**Comment by Munsio** [comment]
For me it is okay if you want to merge #331 first.
The other question is should we wait with this merge-request until i also implemented the I2S to the trinket or should this be a second PR just adding the sound feature.

**Comment by deadprogram** [comment]
It will probably require some additions to get it to compile with the I2S changes. Getting it to work with a separate mic is another story. As long as it compiles for now would be enough for me.

**Comment by deadprogram** [comment]
@Munsio the I2S PR was merged. Can you please rebase `dev` into this branch, and make sure that your example still builds as you expect? Thank you.

**Comment by deadprogram** [comment]
If you rebase dev into your branch, you will discover the following when trying to build your example:

```
$ tinygo build -size short -o test.elf -target=trinket-m0 examples/blinky1
# machine
src/machine/machine_atsamd21.go:678:16: undeclared name: I2S_SCK_PIN
src/machine/machine_atsamd21.go:679:15: undeclared name: I2S_WS_PIN
src/machine/machine_atsamd21.go:680:15: undeclared name: I2S_SD_PIN
```

This is what I was referring to in https://github.com/tinygo-org/tinygo/pull/333#issuecomment-490826035

**Comment by deadprogram** [comment]
Thank you very much for all the great work on this @Munsio now merging. Take a victory lap!
